docs/LgcOverview.md

```diff
@@ -0,0 +1,199 @@
+and add metadata.
+
+### BuilderContext
+
+An object of this class is created first, and it spans multiple compiles. It contains state
+that spans multiple compiles: information on the target GPU. It has an LLVMContext attached,
+and, like LLVMContext, you cannot use the same one for concurrent compiles.
+
+BuilderContext provides methods to create Pipeline and Builder.
+
+### Pipeline
+
+A Pipeline object represents a single pipeline compilation. It is not created at all for a single
+shader compilation. It contains methods to
+* set pipeline state, including pipeline and per-shader tuning options;
+* link individual shader IR modules into the pipeline module;
+* generate (run the middle-end and back-end parts of the compilation process).
+
+### Builder
+
+Builder is a subclass of LLVM's IRBuilder<>, so the front-end can use all of its methods to
+generate IR at the current insertion point and with the current debug location and (in some cases)
+fast math flags.
+
+In addition, Builder contains methods to generate IR for graphics-pipeline-specific constructs,
+such as buffers, images, and shader inputs and outputs.
+
+Builder also contains methods to set "shader modes", that is, additional modes specified in
+the shader input language (SPIR-V) such as floating point mode, tessellation mode, workgroup
+size, etc. These are set via Builder rather than Pipeline because they need to be set even
+in the case of compiling a single shader; in that case the modes are saved in its IR for future
+linking and generating.
+
+The front-end decides which one of two modes Builder works in, although which mode is selected
+does not otherwise affect the front-end:
+
+* BuilderRecorder, where methods to generate IR for graphics-pipeline-specific constructs are
+  "recorded" in the IR by a call to `@llpc.call.*` for each one, and then replayed at the start
+  of the middle-end;
+
+* BuilderImpl, where those same methods generate IR more directly.
+
+If doing a single shader compile, BuilderRecorder is always used.
+
+### Typical front-end flow
+
+There are two different ways of compiling:
+
+* A pipeline compile, where the front-end compiler has the pipeline state and the individual shaders
+  at the same time.
+
+* A shader compile, where the front-end compiler has only one shader, without yet knowing how it relates
+  to any pipelines. Here, the front-end compiles the shader using Builder, and then, later on, there
+  is a pipeline compile where the input shaders are IR from the individual shader compiles.
+
+The front-end flow is:
+
+* Call `BuilderContext::Initialize`. The first time this is called must be before LLVM command-line
+  processing, and before the first call to `BuilderContext::Create`. It is allowed to call it multiple
+  times.
+
+* Process command-line options with `cl::ParseCommandLineOptions`. This can only be done once in the
+  whole process.
+
+* Create a BuilderContext using `BuilderContext::Create`, passing in
```

Comment:
It seems this section isn't well formatted, each paragraph has a *, is it expected? AsciidocFX is used in my local machine.

Reply:
You can see the formatted markdown in github by browsing the commit and selecting the md file. Github then shows it with formatting. E.g.
https://github.com/trenouf/llpc/blob/pr380-overview/docs/LgcOverview.md

Using that view, the formatting looks ok to me.

Reply:
got it. please ignore my previous comments. thanks!